### PR TITLE
Track ammo for small support vehicles

### PR DESCRIPTION
**Pull request by neoancient**
Small support vehicles (under 5 tons) use the same weapons as conventional infantry. Unlike infantry, they track ammunition expenditures. Rather than create a new AmmoType instance for each type of ranged infantry weapon, I created one for standard infantry ammo and one for inferno ammo, which is an alternate munition type available to certain weapons. This is possible because the ammo occupies the same slot as the weapon and cannot be shared between weapons. So the infantry ammo is the same type as the weapon it's linked to.

The ammo mount itself is in LOC_NONE, like one-shot ammo. Weapons that can use either standard or inferno ammo have two bins in a chain. When switching munition type their positions in the chain are swapped. This is the same approach used by the protomech fusillade launcher, which is basically a one-shot weapon except it has two shots and they don't have to be the same munition type. The difference with small SV ammo is that each bin can have multiple shots. Weapons with alternate inferno ammo are implemented as two separate EquipmentType instances. I went through the weapons and changed the internalNames so that the internalName of the inferno variant is always the standard variant with "Inferno" appended. Previous internalNames were added as lookup names. This makes it easy to tell whether the weapon has an inferno variant and to find it, which is necessary for using the right damage values for the ammo type.

Unlike heavy weapon ammo, which comes in one-ton (or half-ton) lots with a certain number of shots per ton, small and medium weapon ammo comes in clips that vary in weight and number of shots from weapon to weapon. This is coded in the weapon class itself.

Edit: Here's a unit I threw together for testing inferno ammo.

[Firebug (Standard).blk.txt](https://github.com/MegaMek/megamek/files/5215915/Firebug.Standard.blk.txt)

### CODE REVIEW

**Review by NickAragua — COMMENTED** [review]
Looks pretty good, got a couple of questions and one possible suggestion.

File: megamek/src/megamek/common/weapons/infantry/InfantryPistoMachinePistolSPEC7AWeapon.java [hunk 1]
```diff
@@ -1,51 +0,0 @@
-/**
- * MegaMek - Copyright (C) 2004,2005 Ben Mazur (bmazur@sev.org)
```
[the remark this answers is not in the document]
Comment (neoancient) [reply megamek/src/megamek/common/weapons/infantry/InfantryPistoMachinePistolSPEC7AWeapon.java:2]: I don't remember deleting it, but it looks like it's a duplicate of InfantryPistolSMGGHTSpec7aWeapon, so I probably kept the one that was spelled correctly.

File: megamek/src/megamek/common/weapons/infantry/InfantryPistolISPulseLaserPistolWeapon.java [hunk 2]
```diff
@@ -36,13 +36,16 @@ public InfantryPistolISPulseLaserPistolWeapon() {
...
 		addLookupName("Pulse Laser Pistol");
-		ammoType = AmmoType.T_NA;
+		ammoType = AmmoType.T_INFANTRY;
 		cost = 1000;
 		bv = 0.13;
 		tonnage = .001;
 		flags = flags.or(F_NO_FIRES).or(F_DIRECT_FIRE).or(F_ENERGY);
 		infantryDamage = 0.14;
 		infantryRange = 0;
+		ammoWeight = 0.0003;
```
[the remark this answers is not in the document]
Comment (neoancient) [reply megamek/src/megamek/common/weapons/infantry/InfantryPistolISPulseLaserPistolWeapon.java:46]: The infantry weapon tables in TM.

File: megamek/src/megamek/common/MechView.java [hunk 3]
```diff
@@ -911,6 +912,30 @@ private ViewElement getAmmo() {
...
+                        entity.getLocationAbbr(mounted.getLocation()),
+                        String.valueOf((int) mounted.getSize() * ((InfantryWeapon) mounted.getType()).getShots()),
+                        ""};
+                if (entity.isOmni()) {
+                    row[3] = mounted.isOmniPodMounted() ?
+                            Messages.getString("MechView.Pod") : //$NON-NLS-1$
+                            Messages.getString("MechView.Fixed"); //$NON-NLS-1$
+                }
+                int shotsLeft = 0;
+                for (Mounted current = mounted.getLinked(); current != null; current = current.getLinked()) {
```
[the remark this answers is not in the document]
Comment (neoancient) [reply megamek/src/megamek/common/MechView.java:927]: As mentioned in the PR description, the way I handled alternate munitions that can only be used by one weapon is to chain link them. So weapons that have alternate inferno ammo get two bins.

[the remark this answers is not in the document]
Comment (neoancient) [reply megamek/src/megamek/common/MechView.java:927]: This code is agnostic. It works regardless of the number of bins linked to the weapon.

File: megamek/src/megamek/common/templates/SupportVeeTROView.java [hunk 4]
```diff
@@ -191,8 +192,21 @@ protected int addEquipment(Entity entity) {
                 }
                 fields.put("tonnage", adjustWeight(eq.getTonnage(entity, entry.getKey().getSize()) * count));
                 fields.put("location", loc);
-                fields.put("slots", eq.getCriticals(entity, entry.getValue()) * count);
+                fields.put("slots", eq.getSupportVeeSlots(entity) * count);
                 weaponList.add(fields);
+                if (eq instanceof InfantryWeapon) {
+                    Map<String, Object> ammoFields = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (neoancient) [reply megamek/src/megamek/common/templates/SupportVeeTROView.java:198]: The method in TROView never gets called for support vehicles. I think what happened is that I started reworking TROView then realized that some differences in logic would require overriding it, but never removed the extra code from the parent class. This is what comes of working in 10-15 minute sessions with several days between.

**Review by NickAragua — APPROVED** [review]

**Pull request merged**